pkg/3scale/amp/operator/redis_reconciler.go

```diff
@@ -124,3 +147,100 @@ func Redis(apimanager *appsv1alpha1.APIManager, client client.Client) (*componen
 	}
 	return component.NewRedis(opts), nil
 }
+
+func (r *RedisReconciler) deleteSystemRedisSecretNamespaceAttribute() (reconcile.Result, error) {
+	redis, err := Redis(r.apiManager, r.Client())
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	desired := redis.SystemRedisSecret()
+	existing := &corev1.Secret{}
+	err = r.Client().Get(context.TODO(), types.NamespacedName{Name: desired.Name, Namespace: r.apiManager.Namespace}, existing)
+	if k8serr.IsNotFound(err) {
+		return reconcile.Result{}, nil
+	}
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	update := false
+	namespaceAttr := "NAMESPACE"
+	if _, ok := existing.Data[namespaceAttr]; ok {
+		update = true
+		delete(existing.Data, namespaceAttr)
+	}
+	if update {
```

Comment:
I think we don't need `update` var here, if it's existing > delete > update.

Reply:
ReChecked. Not removing without this. Thank you

Reply:
I think I have maybe worded it incorrectly. 
Doing this kind of style update is good when you need to remove multiple things but you don't know if these things are present. For example, if you remove "NAMESPACE" "NAMESPACE1" "NAMESPACE2" - you use for loop to go through them and find them, if found, you remove and set update to true. 
Then, you don't have to do multiple updates but one based on the "update" env. 
In this case, we are only removing one key - "NAMESPACE" - therefore the extra logic of having additional var to indicate that update is required is not needed. The code could look something like this:
```
	if _, ok := secret.Data["NAMESPACE"]; ok {
		delete(secret.Data, namespaceKey)
		err = r.UpdateResource(secret)
		if err != nil {
			return err
		}
	}
```

Reply:
oh, yes, sure. Done. Thanks